vendor/bat-native-ads/src/bat/ads/internal/ads_impl.cc

```diff
@@ -149,6 +154,11 @@ void AdsImpl::InitializeStep4(
 
   NotificationAllowedCheck(false);
 
+#if defined(OS_ANDROID)
```

Comment:
Please use run-time checks, i.e. `IsMobile` or add specific `IsAndroid` and `IsIOS` functions as this allows fast sanity testing of business logic for both desktop and mobile

Reply:
@tmancey 
It's not clear how do you plan to test android-specific logic on desktop. Please provide  an example.

Reply:
For example by changing `IsMobile` to return `true` I am able to test notifications, catalog etc. business logic as it is different on mobile to desktop without the need to build all platforms. Once dev is complete on a mobile/desktop change then I sanity test on Android or Desktop etc.

Reply:
After discussing with @bridiver `IsMobile` would be refactored to use feature checks, see https://github.com/brave/brave-browser/issues/6229. However the above change would still need to be made

Reply:
NONCRITICAL
Covered by: brave/brave-browser#6235 (7)